src/app/(auth)/register/_components/RegisterForm.tsx

```diff
@@ -0,0 +1,244 @@
+"use client";
+import { Input, Label } from "@/components/shared";
+import { cn } from "@/lib/utils";
+import { useForm } from "react-hook-form";
+import Link from "next/link";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+    Form,
+    FormControl,
+    FormDescription,
+    FormField,
+    FormItem,
+    FormLabel,
+    FormMessage,
+} from "@/components/ui/form";
+
+const formSchema = z
+    .object({
+        firstname: z.string().min(1, { message: "Invalid name" }).max(30),
+        lastname: z.string().min(1, { message: "Invalid lastname" }).max(30),
+        tel: z
+            .string()
+            .refine(
+                (value) =>
+                    /^[+]{1}(?:[0-9-()/.]\s?){6,15}[0-9]{1}$/.test(value),
+                { message: "Invalid number" },
+            ),
+        email: z
+            .string()
+            .email({ message: "Invalid email address" })
+            .min(1, { message: "required_error" }),
+        password: z.string().min(6, "Password is too short"),
+        confirmpassword: z.string().min(6, "Password is too short"),
+    })
+    .refine((data) => data.password === data.confirmpassword, {
+        message: "Passwords don't match",
+        path: ["confirmpassword"],
+    });
+
+export const RegisterForm = () => {
```

Comment:
add prop onSubmit and prop type via React.FC
onSubmit type is values: z.infer<typeof formSchema>